### PR TITLE
Add customizable timestamps

### PR DESCRIPTION
**Pull request by Trevor159**
I wanted a config option for seconds, I thought this was better. Added year, month, day simply because I was already making it customize-able not because I thought they were useful. Someone might want that who knows.

EDIT: replaces #6955

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampConfig.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+
+@ConfigGroup("timestamp")
+public interface TimestampConfig extends Config
+{
+	@ConfigItem(
+			keyName = "format",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampConfig.java]: Fix your indentation: <https://github.com/runelite/runelite/wiki/Code-Conventions>

File: runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampConfig.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+public interface TimestampConfig extends Config
+{
+	@ConfigItem(
+		keyName = "format",
+		name = "Timestamp Format",
+		description = "Customize your timestamp format by using the following characters<br>" +
+				"'%Y' : year<br>" +
+				"'%MO' : month<br>" +
+				"'%D' : day<br>" +
+				"'%H24' : hour in military time<br>" +
```
Comment (Spedwards) [inline runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampConfig.java]: It's called 24-hour time, not military time. The military just uses it.

File: runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampConfig.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		name = "Timestamp Format",
+		description = "Customize your timestamp format by using the following characters<br>" +
+				"'%Y' : year<br>" +
+				"'%MO' : month<br>" +
+				"'%D' : day<br>" +
+				"'%H24' : hour in 24 hour format<br>" +
+				"'%H12' : hour in 12 hour format<br>" +
+				"'%MI' : minute<br>" +
+				"'%S' : second",
+		position = 0
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampConfig.java]:
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampPlugin.java [hunk 4]
```diff
@@ -76,10 +87,14 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 		final ZonedDateTime time = ZonedDateTime.ofInstant(
 			Instant.ofEpochSecond(messageNode.getTimestamp()), ZoneId.systemDefault());
 
-		final String dateFormat = time.get(ChronoField.HOUR_OF_DAY) + ":" +
-			String.format("%02d", time.get(ChronoField.MINUTE_OF_HOUR));
-
-		String timestamp = "[" + dateFormat + "] ";
+		String timestamp = config.timestampFormat()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampPlugin.java]:
```suggestion
		final String timestamp = config.timestampFormat()
```

File: runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampConfig.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+		keyName = "format",
+		name = "Timestamp Format",
+		description = "Customize your timestamp format by using the following characters<br>" +
+				"'%Y' : year<br>" +
+				"'%MO' : month<br>" +
+				"'%D' : day<br>" +
+				"'%H24' : hour in 24 hour format<br>" +
+				"'%H12' : hour in 12 hour format<br>" +
+				"'%MI' : minute<br>" +
+				"'%S' : second",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampConfig.java]:
```suggestion
				"'%S' : second"
```

File: runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampPlugin.java (hunk 4 above)
Comment (15987632) [reply runelite-client/src/main/java/net/runelite/client/plugins/timestamp/TimestampPlugin.java]: you probably don't really care anymore but this was wrong

**Comment by Trevor159** [comment]
rebased

**Comment by Adam-** [comment]
Does this account for timezones on the host system?

**Comment by deathbeam** [comment]
SimpleDateFormat is using default system timezone if none is specified.

**Comment by Adam-** [comment]
This works too:

```java
        SimpleDateFormat formatter = new SimpleDateFormat("[HH:mm]");

        final ZonedDateTime time = ZonedDateTime.ofInstant(
            Instant.ofEpochSecond(System.currentTimeMillis()), ZoneId.systemDefault());

        String timestamp = formatter.format(Date.from(time.toInstant())) + " ";
        System.out.println(timestamp);```

**Comment by Trevor159** [comment]
@Adam- I think I did what you asked but the test I have now is obviously wrong. It fails if the timezone is not my own (est). I'm not sure how I could get the test to work without specifying a timezone in the test which would defeat the whole point of the test I think.

**Pull request merged**